src/core/account/Account.ts

```diff
@@ -0,0 +1,245 @@
+ * In order to use the SingleKey authentication scheme, `legacy` needs to be explicitly set to false.
+ */
+export interface CreateEd25519AccountFromPrivateKeyArgs {
+  privateKey: Ed25519PrivateKey;
+  address?: AccountAddressInput;
+  legacy?: true;
+}
+
+/**
+ * Arguments for creating an `SingleKeyAccount` from an `Ed25519PrivateKey`.
+ * The `legacy` argument needs to be explicitly set to false in order to
+ * use the `SingleKey` authentication scheme.
+ */
+export interface CreateEd25519SingleKeyAccountFromPrivateKeyArgs {
+  privateKey: Ed25519PrivateKey;
+  address?: AccountAddressInput;
+  legacy: false;
+}
+
+/**
+ * Arguments for creating an `SingleKeyAccount` from any supported private key
+ * that is not an `Ed25519PrivateKey`.
+ * The `legacy` argument defaults to false and cannot be explicitly set to true.
+ */
+export interface CreateSingleKeyAccountFromPrivateKeyArgs {
+  privateKey: Exclude<PrivateKey, Ed25519PrivateKey>;
+  address?: AccountAddressInput;
+  legacy?: false;
+}
+
+/**
+ * Arguments for creating an opaque `Account` from any supported private key.
+ * This is used when the private key type is not known at compilation time.
+ */
+export interface CreateAccountFromPrivateKeyArgs {
+  privateKey: PrivateKey;
+  address?: AccountAddressInput;
+  legacy?: boolean;
+}
+
+/**
+ * Arguments for generating an `Ed25519Account`.
+ * This is the input type used by default.
+ */
+export interface GenerateEd25519AccountArgs {
+  scheme?: SigningSchemeInput.Ed25519;
+  legacy?: true;
+}
+
+/**
+ * Arguments for generating an `SingleKeyAccount` with ah underlying `Ed25519PrivateKey`.
+ * The `legacy` argument needs to be explicitly set to false,
+ * otherwise an `Ed25519Account` will be returned instead.
+ */
+export interface GenerateEd25519SingleKeyAccountArgs {
+  scheme?: SigningSchemeInput.Ed25519;
+  legacy: false;
+}
+
+/**
+ * Arguments for generating an `SingleKeyAccount` with any supported private key
+ * that is not an `Ed25519PrivateKey`.
+ * The `legacy` argument defaults to false and cannot be explicitly set to true.
+ */
+export interface GenerateSingleKeyAccountArgs {
+  scheme: Exclude<SigningSchemeInput, SigningSchemeInput.Ed25519>;
+  legacy?: false;
+}
+
+/**
+ * Arguments for generating an opaque `Account`.
+ * This is used when the input signature scheme is not known at compilation time.
+ */
+export interface GenerateAccountArgs {
+  scheme?: SigningSchemeInput;
+  legacy?: boolean;
+}
+
+/**
+ * Arguments for deriving a private key from a mnemonic phrase and a BIP44 path.
+ */
+export interface PrivateKeyFromDerivationPathArgs {
+  path: string;
+  mnemonic: string;
+}
+
+/**
+ * Interface for a generic Aptos account.
+ *
+ * The interface is defined as abstract class to provide a single entrypoint for account generation,
+ * either through `Account.generate()` or `Account.fromDerivationPath`.
+ * Despite this being an abstract class, it should be treated as an interface and enforced using
```

Comment:
Dependency cycles 😢

Reply:
Sad

Reply:
`Account` needs full imports for all accounts implementations.
If you do `class MyAccount extends Account` you need full import from `Account`

Reply:
so the options are:
1) move the generalized constructors outside of the `Account` class and into a separate file
2) don't use `extends` (which is acceptable since there's no particular benefit in doing that)
3) putting everything in the same file, which is not sustainable

To me option 1 would be the best one, but I got pushbacks since "users won't be able to easily find how to create an account".

Another option would be to do 1, and then use mixins.

Reply:
if we want to change this up in the future, it should be a quick fix